PublicMapping/districtbuilder
# Updates to support CVAP data

## Overview

When trying to add CVAP and VAP data to all 50 states, it predictably increased file size.
To work around this, I removed the `demographics` and `voting` properties from the TopoJSON in #1197 

For reasons unknown to me, even after dropping these fields the memory used by the server seems to be much higher than for the previous TopoJSON data, to the point that the server would get killed by running out of memory before loading even half of all states.

To determine a fix, I did several things:
 
 Added a script to load full state data locally, to ensure the cache settings I had would work for a local development instance or a small server like we'd like to use for staging https://github.com/PublicMapping/districtbuilder/commit/03ca0e9bad00a960b9098bc66d9c7ba82781174d
 - This revealed some race conditions that were only present when loading many states from disk at once (something that doesn't happen in production) which I fixed with https://github.com/PublicMapping/districtbuilder/commit/336bece12806cd1b369c9fc37ca55b532bc49656 and https://github.com/PublicMapping/districtbuilder/commit/ba6721f8635ca8d1adde42bab3d96c4d14975b07
 - Having that many states loaded at once makes it really painful if you accidentally run the server without realizing it, so I changed some of our scripts to use the `--no-deps` flag for `docker-compose run` https://github.com/PublicMapping/districtbuilder/commit/f56dc35f0c9c60f32b317a8287e78b31d4d83ef6

I tried playing with the max cache size for a bit, but didn't get very far here - I would have had to reduce it down far enough that we could no longer load all 50 states on a 32Gb instance.

I decided to try using a different serialization format again, which had much better success, and very different performance characteristics from when I last looked at things (https://github.com/PublicMapping/districtbuilder/issues/1142#issuecomment-1049286088), presumably due to dropping a lot of data fields.

You can see [this Gist for the benchmarks I wrote](https://gist.github.com/maurizi/dc78d915c7aace64c33aa554934f541b) - the summary is that `v8` deserialization is no longer performing well for us, and we would be better served by either `JSON.stringify` or something custom like `topobuf`.
Since after dropping unneeded data fields we're now below the 500Mb limit on string length for all of the production TopoJSON files, I switched us back to using `JSON.stringify`. This is primarily for the sake of simplicity, though `topobuf` is only slightly slower and does have lower memory requirements and a larger maximum file size, so we could switch to that in the future to support a larger region like the custom PA for the PA LRC organization that exceeds the file size limitations of `JSON.stringify`.

While I was waiting for CI to provision repeatedly I also worked on a refactor to `worker-pool.ts` to turn it into a NestJS service to be more in line with code standards on the rest of the server: https://github.com/PublicMapping/districtbuilder/commit/f25de63212c1fb4e3029699445bab40abbe5aaec

### Checklist


### Demo

Staging is running with CVAP & VAP data right now 😃 

## Testing Instructions

- `scripts/update`
- `scripts/load-full-state-data`
- Your server should be able to start (and should load relatively quickly)
- Remove your cached TopoJSON data and restart your server `sudo rm -rf /var/cache/district-builder-server-topology/*` - it should still load, though slower as it will need to download the files again

Connects to #1171
